### PR TITLE
Use kubernetes client-go

### PR DESCRIPTION
**Pull request by galal-hussein**
Changes include:

- Replace calls to kubernetes api in kubernetesclient package with client-go clientset calls.
- Replace watching events using websocket from kubernetes client with client-go informer watch.
- Change all kubernetes-model resources with client-go resources.
- modifying unit tests to use client-go watches instead of kubernete-model watchevents.

### CODE REVIEW

**Comment by galal-hussein** [comment]
@alena1108 @joshwget RFR

File: watchevents/generic.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	return genHandler
+}
+
+func (g *genericHandler) startGenericWatch(resourceName string) chan struct{} {
+	runtimeObj, restClient := g.getRuntimeObject(resourceName)
+	watchlist := cache.NewListWatchFromClient(restClient, resourceName, "", fields.Everything())
+	_, controller := cache.NewInformer(
+		watchlist,
+		runtimeObj,
+		time.Second*0,
```
Comment (joshwget) [inline watchevents/generic.go:40]: @alena1108 Should any of the watches set a resync period?

File: hostlabels/syncer.go [hunk 2]
```diff
@@ -68,20 +68,20 @@ func sync(kClient *kubernetesclient.Client, metadataClient metadata.Client, c *c
...
 				break
 			}
-			v2, ok := node.Metadata.Labels[k]
+			v2, ok := node.Labels[k]
 			if !ok {
 				// This label doesn't exist
 				changed = true
 			} else {
-				v2String, ok := v2.(string)
+				v2String := v2
```
Comment (alena1108) [inline hostlabels/syncer.go:76]: no need for v2String here. Having v2 is enough, it is a string already

File: kubernetesclient/authorization.go [hunk 3]
```diff
@@ -1,50 +1,27 @@
...
 	}
-	token = strings.TrimSpace(string(bytes))
-	if token == "" {
-		return errors.New("No token passed in from stdin")
-	}
-
-	caData, err = ioutil.ReadFile(caLocation)
 	if err != nil {
-		return fmt.Errorf("Failed to read CA cert %s: %v", caLocation, err)
+		panic(err.Error())
```
Comment (alena1108) [inline kubernetesclient/authorization.go:20]: no panic! :) Use github.com/Sirupsen/logrus.fatal construct

File: kubernetesclient/client.go [hunk 4]
```diff
@@ -1,27 +1,17 @@
...
-
-const byNamePath string = "/api/v1/namespaces/%s/%s/%s"
+import "k8s.io/client-go/kubernetes"
 
 func NewClient(apiURL string, debug bool) *Client {
 	client := &Client{
 		baseClient: baseClient{
-			BaseURL: apiURL,
-			debug:   debug,
+			K8sClientSet: GetK8sClientSet(apiURL),
```
Comment (alena1108) [inline kubernetesclient/client.go:8]: I am not keen on the `K8sClientSet' naming. And can't we use the k8s client directly, w/o wrapping it in baseClient? Is debug field ever being used? If not, lets get rid of it

**Comment by alena1108** [comment]
@galal-hussein do we ever use rc_client, pod_client? If not, and only node_client is being used, lets remove the rest.

File: scripts/test [hunk 5]
```diff
@@ -70,6 +70,28 @@ until $(curl --output /dev/null --silent --fail http://localhost:8080/api/v1/ser
...
 
+mkdir -p /etc/kubernetes/ssl/
+cat > /etc/kubernetes/ssl/kubeconfig << EOF
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    api-version: v1
+    insecure-skip-tls-verify: true
+    server: "http://localhost:8080"
```
Comment (alena1108) [inline scripts/test]: Localhost? Shouldn't it be a kubernetes URL?

**Comment by alena1108** [comment]
@galal-hussein please split the PR into 2 commits - one with vendor changes, another having all the code changes.

File: watchevents/generic.go (hunk 1 above)
Comment (alena1108) [reply watchevents/generic.go:40]: @galal-hussein looks like we can drop generic events handling. It was used for old UI only, we do not need it anymore.

**Comment by galal-hussein** [comment]
closed in favor of #52

**Pull request closed without merging**